version: "3.9"
services:
  app-takabase-api:
    platform: linux/arm64/v8
    build:
      context: .
      dockerfile: Dockerfile
      target: development
    depends_on:
      app-takabase-mysql:
        condition: service_healthy
    container_name: app-takabase-api
    restart: unless-stopped
    tty: true
    env_file:
      - .env.takabase-local
    environment:
      APP_PORT: 8080
      APP_HOST: 0.0.0.0
    ports:
      - "8080:8080"
      - "4400:4400"
      - "5555:5555"
      - "9229:9229"
    volumes:
      - .:/usr/src/app/
    networks:
      - api-takabase-network
    command: npm run start:dev
  app-takabase-mysql:
    platform: linux/arm64/v8
    image: mysql:8
    container_name: app-takabase-mysql
    restart: unless-stopped
    tty: true
    env_file:
      - .env.takabase-local
    environment:
      MYSQL_ROOT_PASSWORD: root
    healthcheck:
      test: mysqladmin ping -h 127.0.0.1 -u ${API_MYSQL_USER} --password=${API_MYSQL_PASSWORD}
      interval: 3s
      timeout: 5s
      retries: 10
    volumes:
      - app-takabase-mysql-data:/var/lib/mysql/
      - ./src/database/mysql:/docker-entrypoint-initdb.d/
    networks:
      - api-takabase-network

networks:
  api-takabase-network:
    driver: bridge

volumes:
  app-takabase-mysql-data:
    driver: local
